contracts/NFTController.sol

```diff
@@ -1,10 +1,48 @@
 // SPDX-License-Identifier: MIT
 
 pragma solidity ^0.8.0;
+import "./Mocks/IERC721.sol";
 
+contract NFTController {
+IERC721 public nonFungibleContract;
 
-contract Controller {
+  struct Auction {
+    address seller;
+    uint128 price;
```

Comment:
we may want the auction to just have an end date, rather than a set price. so we could prob add another parameter to `createAuction` with an `endTime`